gt/main.c

```diff
@@ -90,30 +186,72 @@ gt_proc(void *arg)
 			 *
 			 * Other packets will be fowarded directly.
 			 */
+			rte_pktmbuf_adj(m, sizeof(struct ether_hdr));
+
+			ip_hdr = get_ip_hdr(m, &inner_ip_ver);
```

Comment:
As Cody has already pointed out, you need to squash the decapsulation patch with this patch because `get_ip_hdr()` is reading the outer IP header, that is, the IP header of the tunnel, and the policy is on the inner IP header.